test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/IVsServicesTestCase.cs

```diff
@@ -80,5 +82,237 @@ public void IVSPathContextProvider2_WithEmptySolution_WhenTryCreateUserWideConte
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""PrivateRepository"" value=""{privateRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""PrivateRepository"">
+            <namespace id=""Contoso.*"" />             
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+
+            solutionService.CreateEmptySolution("TestSolution", solutionDirectory);
+            var projExt = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V46, "TestProject");
+            solutionService.SaveAll();
+            var project = dte.Solution.Projects.Item(1);
+
+            // Act
+            nugetTestService.InstallPackage(project.UniqueName, "contoso.a", "1.0.0");
+            nugetTestService.InstallPackage(project.UniqueName, "contoso.a", "2.0.0");
+
+            // Assert
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, projExt, packageName, XunitLogger);
+
+            var packagesDirectory = Path.Combine(solutionDirectory, "packages");
+            var uniqueContentFile = Path.Combine(packagesDirectory, packageName + '.' + packageVersionV2, "lib", "net5.0", "Thisisfromprivaterepo2.txt");
+            // Make sure version 2 is restored.
+            Assert.True(File.Exists(uniqueContentFile));
+        }
+
+        [StaFact]
+        public async Task SimpleInstallFromIVsInstaller_PackageNamespace_WithMultipleFeedsWithIdenticalPackages_InstallsCorrectPackage()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+            var dte = VisualStudio.Dte;
+            string mainDirectory = CommonUtility.CreateSolutionDirectory(Directory.GetCurrentDirectory());
+            string solutionDirectory = Path.Combine(mainDirectory, "Solution");
+            var solutionService = VisualStudio.Get<SolutionService>();
+            var nugetTestService = GetNuGetTestService();
+
+            var packageName = "Contoso.A";
+            var packageVersionV1 = "1.0.0";
+            var packageVersionV2 = "2.0.0";
+
+            var opensourceRepositoryPath = Path.Combine(mainDirectory, "OpensourceRepository");
+            Directory.CreateDirectory(opensourceRepositoryPath);
+
+            await CommonUtility.CreateNetCorePackageInSourceAsync(opensourceRepositoryPath, packageName, packageVersionV1, "Thisisfromopensourcerepo1.txt");
+            await CommonUtility.CreateNetCorePackageInSourceAsync(opensourceRepositoryPath, packageName, packageVersionV2, "Thisisfromopensourcerepo2.txt");
+
+            var privateRepositoryPath = Path.Combine(mainDirectory, "PrivateRepository");
+            Directory.CreateDirectory(privateRepositoryPath);
+
+            await CommonUtility.CreateNetCorePackageInSourceAsync(privateRepositoryPath, packageName, packageVersionV1, "Thisisfromprivaterepo1.txt");
```

Comment:
The rest of the code looks reasonable, but I don't understand why we're creating netcore packages when all the projects can only be .NET Framework.

Reply:
Ok. Addressed it.